networkx/readwrite/gml.py

```diff
@@ -311,7 +311,8 @@ def tokenize():
                         if i == 0:  # keys
                             value = group.rstrip()
                         elif i == 1:  # reals
-                            value = float(group)
+                            value = group.replace("_", "")  # handle incoming nan
```

Comment:
Would it be better to replace `"_NAN"` with `"NAN"` if the goal is to `handle incoming nan`?

Reply:
Thank you you're welcome!

That makes sense so probably?
Related is whether to have inf go out as INF or +INF and if incoming INF should be float (probably).

Well there is a small collision since NAN is a legal key, and labels can be keys.
NetworkX allows labels that are unquoted floats and converts them to floats.
NetworkX allows labels that are "convertable ASCII value for node id or label".
* this branch is only triggered if the token is a 'key' (unquoted string in "[A-Za-z][0-9A-Za-z_]*\b" )

A. Should the behavior for labels be:
1. Consistent with regular floats: convert NAN and INF to float('nan') and float('inf')
2. Consistent with current behavior: convert NAN and INF to "NAN" and "INF"

B. Should the behavior for keys be:
1. Consistent with GML: interpret NAN and INF as strings
2. Consistent with floats: interpret NAN and INF as float and reject

Reply:
I would prefer to avoid having floats be keys, but that is a bias I have based on round-off difficulties which probably is not relevant for inf/nan.  But my bias is still there based on a principal that using floats as keys can cause confusion.  I guess that means my preference is for strings. But I could probably be convinced otherwise. :}

Reply:
Behavior for labels:   interpret NAN and INF as floats.
Behavior for keys:    interpret NAN and INF as strings.

What do you think?

Reply:
Definitely!  Sorry I haven't been around.  A while back I made the changes required to not use '_' for NAN and to read and write all variants properly but never pushed it.  I will write thorough tests shortly, probably today.